### PR TITLE
test(ui-common): cover buildStatusNotificationPayload 1.6/2.0.x branches (#1834)

### PR DESCRIPTION
**Pull request by jerome-benoit**
## Summary

`ui/common/src/utils/payloadBuilders.ts` exports
`buildStatusNotificationPayload` (introduced by #1834 to share the
OCPP 1.6 vs 2.0.x branching between CLI and Web UI). Its sibling
builders (`buildAuthorizePayload`, `buildStartTransactionPayload`,
`buildStopTransactionPayload`, `buildIdToken`, `isOCPP20x`) all have
`describe` blocks in `ui/common/tests/payloadBuilders.test.ts`, but
`buildStatusNotificationPayload` had none. The S3 regression-focus gate
was therefore vacuous for this builder.

This PR adds a dedicated `describe('buildStatusNotificationPayload')`
block matching the existing file style. Production code in
`payloadBuilders.ts` is **not** touched.

## New cases (9)

| # | Branch    | Scenario                                                                  |
| - | --------- | ------------------------------------------------------------------------- |
| 1 | OCPP 1.6  | explicit `errorCode` → `{connectorId, status, errorCode}`                 |
| 2 | OCPP 1.6  | `errorCode` omitted → payload omits `errorCode` (no `NoError` default)    |
| 3 | OCPP 1.6  | `evseId` pass-through with `errorCode`                                    |
| 4 | OCPP 1.6  | `ocppVersion = undefined` falls through to 1.6 shape                      |
| 5 | OCPP 1.6  | unsupported version (`'3.0'`) → throws via `assertOCPP16OrUndefined`      |
| 6 | OCPP 2.0.1| `evseId` provided → `{connectorId, connectorStatus, evseId}`, no `status`/`errorCode` |
| 7 | OCPP 2.0  | no `evseId` → `{connectorId, connectorStatus}`                            |
| 8 | OCPP 2.0.1| `errorCode` option ignored (1.6-only field)                               |
| 9 | dispatch  | same input, different `ocppVersion` → different shape in the documented way |

OCPP enum constants are used directly
(`OCPP16ChargePointStatus`, `OCPP16ChargePointErrorCode`,
`OCPP20ConnectorStatusEnumType`, `OCPPVersion`) per the AGENTS.md
OCPP-conventions rule "Avoid string literals when an enumeration exists".

## Spec-vs-reality findings (for maintainer review)

While writing the tests against the function's actual behaviour, two
points in the original test plan diverge from what the code does. Per
the task instructions ("verify against the function's actual behaviour,
do NOT invent" and "if you find a real bug while writing tests, STOP,
document it in the PR description"), I am surfacing these without
modifying production code.

### Finding A — `errorCode` omitted (no `NoError` default)

**Plan (test #2):** "OCPP 1.6 with `errorCode` omitted → defaults to
`NoError`."

**Actual (`payloadBuilders.ts`):**

```ts
return {
  connectorId,
  status,
  ...(options?.errorCode != null && { errorCode: options.errorCode }),
  ...(options?.evseId != null && { evseId: options.evseId }),
}
```

When `errorCode` is omitted, the field is **dropped** from the payload;
no default is applied. Test case #2 asserts the actual behaviour
(`errorCode` absent), not the planned default.

If the maintainer's intent is that the simulator-emitted OCPP 1.6
StatusNotification should carry an explicit `errorCode: "NoError"`
(per OCPP 1.6 §6.31 where `errorCode` is required on the wire), this is
a small follow-up worth a separate PR. I did not change behaviour here.

### Finding B — no runtime status validation

**Plan (tests #3 and #5):** "invalid status enum → throws / returns
rejection per the function contract."

**Actual:** the function takes
`status: ChargePointStatus = OCPP16ChargePointStatus | OCPP20ConnectorStatusEnumType`
and passes the value straight into the returned object. The only
runtime-rejection contract is `assertOCPP16OrUndefined(version)`, which
throws for unsupported `ocppVersion` values such as `'3.0'`. There is
no runtime validation of `status` itself — type-level only.

I did not introduce `as any` / `// @ts-ignore` casts to fabricate a
runtime-invalid status (per AGENTS.md type-safety conventions and the
explicit task constraint). Instead, case #5 tests the actual rejection
contract — unsupported version — which is also exercised by the sibling
builder tests (`buildAuthorizePayload`, line 62-67).

If the maintainer wants runtime enum validation for `status` and/or
`errorCode`, that is a behaviour change and belongs in a separate PR.

## Quality gates

| Command                            | Result                                |
| ---------------------------------- | ------------------------------------- |
| `pnpm --filter ui-common typecheck`| exit 0                                |
| `pnpm --filter ui-common lint`     | exit 0                                |
| `pnpm --filter ui-common test`     | 114 pass / 0 fail (was 105, +9 new)   |
| `pnpm --filter ui-common test:coverage` | see below                        |

### Coverage report (lcov, `src/utils/payloadBuilders.ts`)

```
SF:src/utils/payloadBuilders.ts
FN:9,buildAuthorizePayload                FNDA:5
FN:15,buildIdToken                        FNDA:2
FN:19,buildStartTransactionPayload        FNDA:5
FN:44,buildStatusNotificationPayload      FNDA:10
FN:55,buildStopTransactionPayload         FNDA:4
FN:67,isOCPP20x                           FNDA:28
FN:73,assertOCPP16OrUndefined             FNDA:12
FNF:8     FNH:8     (functions:  100%)
BRF:32    BRH:32    (branches:   100%)
LF:162    LH:162    (lines:      100%)
end_of_record
```

`buildStatusNotificationPayload` (line 44) is now hit **10 times**
across the new test cases, fully covering both the OCPP 1.6 and OCPP
2.0.x branches plus the version-dispatch path.

## Out of scope

- No production code change in `payloadBuilders.ts`.
- Spec-vs-reality findings A and B above are documented, not fixed.

## Severity

MAJOR — test hygiene gap closed; S3 regression-focus gate now non-vacuous
for `buildStatusNotificationPayload`.

### CODE REVIEW

File: ui/common/tests/payloadBuilders.test.ts [hunk 1]
```diff
@@ -174,4 +178,138 @@ await describe('payloadBuilders', async () => {
...
+        OCPPVersion.VERSION_16,
+        { errorCode: OCPP16ChargePointErrorCode.NO_ERROR, evseId: 2 }
+      )
+      assert.deepStrictEqual(result, {
+        connectorId,
+        errorCode: OCPP16ChargePointErrorCode.NO_ERROR,
+        evseId: 2,
+        status: OCPP16ChargePointStatus.CHARGING,
+      })
+    })
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply ui/common/tests/payloadBuilders.test.ts]: Addressed in 11a4ab8b. Test renamed to `'should pass evseId through for OCPP 1.6 without errorCode'` and now passes only `options.evseId` (no `errorCode`), asserting `errorCode` is absent from the resulting payload.

Rationale for the minimal split: the `errorCode`-only path is already independently covered by `'should build OCPP 1.6 payload with status and explicit errorCode'`, so adding back a redundant combined-flag test would just duplicate the spread-syntax composition. The two single-flag tests + the function's spread syntax cover the combined case by construction. Keeps the diff small per AGENTS.md "small, verifiable changes".

Diff: +3/-3 in `ui/common/tests/payloadBuilders.test.ts`. `typecheck`, `lint`, and the 114 ui-common tests all pass locally.

File: ui/common/tests/payloadBuilders.test.ts [hunk 2]
```diff
@@ -174,4 +178,138 @@ await describe('payloadBuilders', async () => {
...
+      assert.throws(
+        () =>
+          buildStatusNotificationPayload(
+            connectorId,
+            OCPP16ChargePointStatus.AVAILABLE,
+            '3.0' as unknown as OCPPVersion
+          ),
+        (error: Error) => error.message.includes('Unsupported OCPP version')
+      )
+    })
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply ui/common/tests/payloadBuilders.test.ts:249]: Declining this one. Two reasons:

1. **Behavioural premise is incorrect.** Per the [Node.js docs](https://nodejs.org/api/assert.html#assertthrowsfn-error-message), when a validator function returns `false` (or any non-`true` value), `assert.throws` _does_ throw an `AssertionError` — it does **not** silently re-throw the original error. The `.includes(...)` boolean return therefore behaves exactly as intended: `true` accepts the error, `false` triggers the assertion failure. There is no silent-mask behaviour to fix.

2. **Codebase harmonisation.** The exact same `(error: Error) => error.message.includes(...)` pattern is already established on the sibling builder test [`buildAuthorizePayload`](https://github.com/SAP/e-mobility-charging-stations-simulator/blob/test/1834-payload-builders-status-notification-coverage/ui/common/tests/payloadBuilders.test.ts#L62-L67) (lines 62-67) — and the bot itself acknowledges this. AGENTS.md "Naming coherence" / "Follow identified design patterns, naming patterns and coding styles" requires new code to match existing conventions in the same file. Switching this single test to a `RegExp` second argument would create stylistic drift with no behavioural benefit; if the team later prefers `RegExp` here, that is a separate harmonisation pass over both call sites, not an in-scope change for this test-coverage PR.

Keeping the original assertion as-is.
